app/user-data-download/server/cronProcessDownloads.js

```diff
@@ -505,16 +524,21 @@ const continueExportOperation = async function(exportOperation) {
 		}
 
 		if (exportOperation.status === 'downloading') {
-			exportOperation.fileList.forEach((attachmentData) => {
+			const maxSizePerBatch = 10485760; // 10MB
+			const filesBatch = spliceByFileSize(exportOperation.fileList, maxSizePerBatch);
```

Comment:
By the way `spliceByFileSize` is written, won't it return just the first batch?

So if you have [1,2,3,4,5] and to an splice on it (0, 2) it will discard the rest of the elements 👀

Reply:
yes and it's what this pr aims to do. the rationale around this is to ideally not process over 10MBs of data every 2 minutes

Reply:
Understood 👀